### PR TITLE
feat: get asset IDs associated with image 360 

### PR DESCRIPTION
**Pull request by haakonflatval-cognite**
#### Type of change

Feat  

#### Jira ticket :blue_book:

https://cognitedata.atlassian.net/browse/

## Description :pencil:

Add method for getting all assets associated with a 360 image collection. Needed for the next release of Data Explorer

## How has this been tested? :mag:

- Open image set XOM-HCU1_north_v02
- Run this function in e.g. `Image360UI`
- Observe that it has two matches

## Checklist :ballot_box_with_check:

- [x] I am proud of this feature.
- [x] I have performed a self-review of my own code.
- [x] I have added PR type (Feat, Bug, Chore, Test, Docs, Style, Refactor) to the PR title.
- [ ] I have manually tested this for different scenarios (different models, formats, devices, browsers).
- [ ] I have commented my code in hard-to-understand areas.
- [ ] I have made corresponding changes to the public documentation.
- [ ] I have added unit and visuals tests to prove that my feature works to the best of my ability.
- [ ] I have refactored the code for readability to the best of my ability.
- [ ] I have checked that my changes do not introduce regressions in the public documentation.
- [ ] I have outlined any known defects / lacking capabilities in the contents of this PR.
- [ ] I have listed any remaining work that should follow this PR in the description or jira/miro/etc.
- [ ] I have added TSDoc to any public facing changes.
- [ ] I have added "developer documentation" in any package README.md that is related / applicable to this PR.
- [ ] I have noted down and am currently tracking any technical debt introduced in this PR.
- [ ] I have thoroughly thought about the architecture of this implementation.
- [ ] I have asked peers to test this feature.

### CODE REVIEW

File: viewer/packages/data-providers/src/image-360-data-providers/Cdf360ImageEventProvider.ts [hunk 1]
```diff
@@ -332,5 +334,27 @@ export class Cdf360ImageEventProvider implements Image360Provider<Metadata> {
...
+        .list({
+          filter: {
+            annotatedResourceIds: idList.map(id => ({ id })),
+            annotatedResourceType: 'file',
+            annotationType: 'images.AssetLink'
+          }
+        })
+        .autoPagingToArray();
+      const assetIds = annotationArray.map(
+        annotation => (annotation.data as AnnotationsCogniteAnnotationTypesImagesAssetLink).assetRef
```
Comment (christjt) [inline viewer/packages/data-providers/src/image-360-data-providers/Cdf360ImageEventProvider.ts:350]: Should typeguard here and throw with proper error message if this fails.

File: viewer/packages/data-providers/src/image-360-data-providers/Cdf360ImageEventProvider.ts [hunk 2]
```diff
@@ -332,5 +334,27 @@ export class Cdf360ImageEventProvider implements Image360Provider<Metadata> {
...
+      );
+
+      return assetIds;
+    });
+
+    const assetIds = (await Promise.all(assetListPromises)).flat() as IdEither[];
```
Comment (christjt) [inline viewer/packages/data-providers/src/image-360-data-providers/Cdf360ImageEventProvider.ts:356]: Should typeguard here and throw with proper error message if this fails.

**Review by christjt — APPROVED** [review]
Looks good, only some minor comments

File: viewer/packages/data-providers/src/image-360-data-providers/Cdf360ImageEventProvider.ts (hunk 2 above)
Comment (haakonflatval-cognite) [reply viewer/packages/data-providers/src/image-360-data-providers/Cdf360ImageEventProvider.ts:356]: Instead used `filter()` to only admit elements satisfying `IdEither`.

File: viewer/packages/data-providers/src/image-360-data-providers/Cdf360ImageEventProvider.ts (hunk 1 above)
Comment (haakonflatval-cognite) [reply viewer/packages/data-providers/src/image-360-data-providers/Cdf360ImageEventProvider.ts:350]: :ok_hand:

**Pull request merged**